pkg/apis/agones/v1/gameserver.go

```diff
@@ -221,13 +225,13 @@ func (gss *GameServerSpec) ApplyDefaults() {
 	gss.applyPortDefaults()
 	gss.applyHealthDefaults()
 	gss.applySchedulingDefaults()
-	gss.applyLogLevelDefaults()
+	gss.applySdkServerDefaults()
 }
 
-// applyLogLevelDefaults applies the log level default - "Info"
-func (gss *GameServerSpec) applyLogLevelDefaults() {
-	if gss.Logging.SdkServer == "" {
-		gss.Logging.SdkServer = "Info"
+// applySdkServerDefaults applies the default log level ("Info") for the sidecar
```

Comment:
So @aLekSer added this function in https://github.com/googleforgames/agones/pull/1007 and I modified it, but I'm wondering if we should just remove it. The only reason to explicitly set this field is if we are worried about compatibility and implicit values. There is code in `pkg/sdkserver/sdkserver.go` that defaults the log level to info so doing it here feels redundant and it also means that when you describe a game server you **always** see this field, even if you didn't explicitly set it. 

I think it would be better to just leave the struct empty unless it's explicitly set by a user. WDYT?

Reply:
Yes, I added some redundancy at the early development stage and forgot to remove.
This could be removed now since SDK server is setting default INFO loglevel.

Reply:
So I would actually argue to keep it - basically because when you do a `kubectl get gameserver foo -o yaml` or a `kubectl describe gs foo` you can see what the default value is set to in those commands.

Same as what happens with Pods - you get a bunch more stuff back than what you originally set, because it's the default that has been set by the system.

Reply:
In that case, should we also default in the two port values? The downside of defaulting is that it makes it impossible to tell later if the value was set by the user or defaulted by the system. 

For port values, I was planning to do something along the lines of adding a flag/env variable to the game server container that the SDK can find and set the port to connect to. If we don't default the port values, we can just set this if the user requested a specific port. If we default the values, we will need to always set it. The same applies on the sdkserver sidecar container. Right now, in this PR I only set the flags to configure the two ports if a user specified a non-default port in the gameserver config. If we default the port, then the flags will always be specified, because the values will always be non-nil.